plugins/extractors/presto/presto.go

```diff
@@ -0,0 +1,294 @@
+		Tags:         []string{"oss", "extractor"},
+	}
+}
+
+// Validate validates the configuration of the extractor
+func (e *Extractor) Validate(configMap map[string]interface{}) (err error) {
+	return utils.BuildConfig(configMap, &Config{})
+}
+
+// Init initializes the extractor
+func (e *Extractor) Init(_ context.Context, configMap map[string]interface{}) (err error) {
+	// Build and validate config received from recipe
+	if err = utils.BuildConfig(configMap, &e.config); err != nil {
+		return plugins.InvalidConfigError{}
+	}
+
+	// create presto client
+	if e.client, err = sql.Open("presto", e.config.ConnectionURL); err != nil {
+		return fmt.Errorf("failed to create client: %w", err)
+	}
+
+	if err = e.extractConnectionComponents(e.config.ConnectionURL); err != nil {
+		err = fmt.Errorf("failed to split configs from connection string: %w", err)
+		return
+	}
+
+	return
+}
+
+// Extract collects metadata of the database through emitter
+func (e *Extractor) Extract(_ context.Context, emit plugins.Emit) (err error) {
+	defer e.client.Close()
+
+	catalogs, err := e.getCatalogs()
+	for _, catalog := range catalogs {
+		// Open a new connection to the given catalogs list
+		// to collect schemas information
+
+		db, err := e.connection(catalog)
+		if err != nil {
+			e.logger.Error("failed to connect, skipping catalog", "error", err)
+			continue
+		}
+
+		dbs, err := e.getDatabases(db, catalog)
+		if err != nil {
+			return fmt.Errorf("failed to extract tables from %s: %w", catalog, err)
+		}
+		for _, database := range dbs {
+			tables, err := e.getTables(db, catalog, database)
+			if err != nil {
+				e.logger.Error("failed to get tables, skipping database", "catalog", catalog, "error", err)
+				continue
+			}
+
+			for _, table := range tables {
+				result, err := e.processTable(db, catalog, database, table)
+				if err != nil {
+					e.logger.Error("failed to get table metadata, skipping table", "error", err)
+					continue
+				}
+				// Publish metadata to channel
+				emit(models.NewRecord(result))
+			}
+		}
+	}
+	return nil
+}
+
+// getCatalogs prepares the list of catalogs
+func (e *Extractor) getCatalogs() (list []string, err error) {
+	// Get list of catalogs
+	catalogs, err := e.client.Query("SHOW CATALOGS")
+	if err != nil {
+		return nil, fmt.Errorf("failed to get the list of catalogs: %w", err)
+	}
+
+	var excludeList []string
+	excludeList = append(excludeList, strings.Split(e.config.Exclude, ",")...)
+
+	for catalogs.Next() {
+		var catalog string
+		if err = catalogs.Scan(&catalog); err != nil {
+			return nil, fmt.Errorf("failed to scan schema from %s: %w", catalog, err)
+		}
+		if exclude(excludeList, catalog) {
+			continue
+		}
+		list = append(list, catalog)
+	}
+
+	return list, err
+}
+
+// getDatabases prepares the list of databases
+func (e *Extractor) getDatabases(db *sql.DB, catalog string) (list []string, err error) {
+	// Get list of databases
+	showSchemasQuery := fmt.Sprintf("show schemas in %s", catalog)
```

Comment:
can we make the case for the query consistent?
```suggestion
	showSchemasQuery := fmt.Sprintf("SHOW SCHEMAS IN %s", catalog)
```